doc/maintaining/cli.rst

```diff
@@ -0,0 +1,558 @@
+source directory not CKAN's source directory. For example:
+
+.. parsed-literal::
+
+   |activate|
+   cd |virtualenv|/src/ckanext-spatial
+   ckan -c |/path/to/ckan.ini| foo 
+
+This should not be necessary when using the pre-installed extensions that come
+with CKAN.
+
+Alternatively, you can give the extension's name using the ``--plugin`` option,
+for example
+
+.. parsed-literal::
+
+   ckan -c |/path/to/ckan.ini| --plugin=ckanext-foo foo 
+
+.. todo::
+
+   Running a ckan shell with ``ckan -c |/path/to/ckan.ini|  --plugin=pylons shell``.
+   Useful for development?
+
+Wrong config file path
+======================
+
+AssertionError: Config filename development.ini does not exist
+  This means you forgot to give the ``--config`` or ``-c`` option to tell CKAN
+  where to find your config file. (CKAN looks for a config file named
+  ``development.ini`` in your current working directory by default.)
+
+ConfigParser.MissingSectionHeaderError: File contains no section headers
+  This happens if the config file that you gave with the ``-c`` or ``--config``
+  option is badly formatted, or if you gave the wrong filename.
+
+IOError: [Errno 2] No such file or directory: '...'
+  This means you gave the wrong path to the ``--config`` or ``-c`` option
+  (you gave a path to a file that doesn't exist).
+
+
+-------------------------
+ckan Commands Reference
+-------------------------
+
+The following ckan commands are supported by CKAN:
+
+================= ============================================================
+asset             WebAssets commands.
+config-tool       Tool for editing options in a CKAN config file
+datapusher        Perform commands in the datapusher.
+dataset           Manage datasets.
+datastore         Perform commands to set up the datastore.
+db                Perform various tasks on the database.
+front-end-build   Creates and minifies css and JavaScript files
+generate          Generate empty extension files to expand CKAN
+jobs              Manage background jobs
+less              Compile all root less documents into their CSS counterparts
+minify            Create minified versions of the given Javascript and CSS files.
+notify            Send out modification notifications.
+plugin-info       Provide info on installed plugins.
+profile           Code speed profiler
+search-index      Creates a search index for all datasets
+seed              Create test data in the database.
+server            Start Development server.
+sysadmin          Gives sysadmin rights to a named user.
+tracking          Update tracking statistics.
+translation       Translation helper functions
+user              Manage users.
+views             Create views on relevant resources
+================= ============================================================
+
+
+asset: WebAssets commands
+==================================
+
+Usage::
+
+    asset build            - Builds bundles, regardless of whether they are changed or not
```

Comment:
I think it would be helpful to put the full command-line for these commands e.g.
```suggestion
    ckan asset build            - Builds bundles, regardless of whether they are changed or not
```
For brevity I think we can get away with not specifying the `-c` parameter in this .rst file, because it will prompt you for that, and it's explained earlier in this .rst.